model/flow/identifier.go

```diff
@@ -64,6 +65,17 @@ func (id Identifier) Format(state fmt.State, verb rune) {
 	}
 }
 
+// IsSampled is a utility method to sample entities based on their ids
+func (id Identifier) IsSampled(sensitivity uint) bool {
+	if sensitivity > 64 {
+		return false
+	}
+	// take the first 8 bytes and check the first few bits based on sensitivity
+	// higher sensitivity means more bits has to be zero, means less number of samples
+	// sensitivity of zero, means everything is sampled
+	return binary.BigEndian.Uint64(id[:8])>>uint64(64-sensitivity) == 0
+}
+
```

Comment:
Better to let user define the sample rate with percentage instead of sensitivity

```suggestion

// IsSampledWithPercent is a utility method to sample entities based on their ids, the sampling rate is 
// percentage based.
// 0 means 0 data will be collected.
// 100 means all data will be collected.
func (id Identitifier) IsSampledWithPercent(percentage uint) bool {
  sensitivity := percentageToSensitivity(percentage)
  return IsSample(sensitivity)
}

func percentageToSensitivity(percentage uint) uint {
  // to be implemented
}
```

Reply:
I can add this as well, but prefer to keep the one used by tracing to not do any extra computation if possible, since this method would be called a lot.